pkg/operation/botanist/controlplane/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -0,0 +1,533 @@
+	SecretName = "kube-controller-manager"
+	// SecretNameServer is the name of the kube-controller-managerr server certificate secret.
+	SecretNameServer = "kube-controller-manager-server"
+
+	// labelRole is a constant for the value of a label with key 'role' whose value is 'controller-manager'.
+	labelRole = "controller-manager"
+	// portNameMetrics is a constant for the name of the metrics port of the kube-controller-manager.
+	portNameMetrics = "metrics"
+
+	// volumeMountPathCA is the volume mount path for the CA certificate used by the kube controller manager.
+	volumeMountPathCA = "/srv/kubernetes/ca"
+	// volumeMountPathCA is the volume mount path for the service account key that is a PEM-encoded private RSA or ECDSA key used to sign service account tokens.
+	volumeMountPathServiceAccountKey = "/srv/kubernetes/service-account-key"
+	// volumeMountPathCA is the volume mount path for the kubeconfig which can be used by the kube-controller-manager to communicate with the kube-apiserver.
+	volumeMountPathKubeconfig = "/var/lib/kube-controller-manager"
+	// volumeMountPathCA is the volume mount path for the x509 TLS server certificate and key for the HTTPS server inside the kube-controller-manager (which is used for metrics and health checks).
+	volumeMountPathServer = "/var/lib/kube-controller-manager-server"
+
+	// managedResourceName is the name of the managed resource that contains the resources to be deployed into the Shoot cluster.
+	managedResourceName = "shoot-core-kube-controller-manager"
+)
+
+// KubeControllerManager contains functions for a kube-controller-manager deployer.
+type KubeControllerManager interface {
+	component.DeployWaiter
+	component.MonitoringComponent
+	// SetSecrets sets the secrets for the kube-controller-manager.
+	SetSecrets(Secrets)
+	// SetReplicaCount sets the replica count for the kube-controller-manager.
+	SetReplicaCount(replicas int32)
+	// WaitForControllersToBeActive checks whether kube-controller-manager has
+	// recently written to the Endpoint object holding the leader information. If yes, it is active.
+	WaitForControllerToBeActive(ctx context.Context) error
+}
+
+// New creates a new instance of DeployWaiter for the kube-controller-manager.
+func New(
+	logger *logrus.Entry,
+	seedClient client.Client,
+	shootClient client.Client,
+	namespace string,
+	version *semver.Version,
+	image string,
+	config *gardencorev1beta1.KubeControllerManagerConfig,
+	podNetwork *net.IPNet,
+	serviceNetwork *net.IPNet,
+) KubeControllerManager {
+	return &kubeControllerManager{
+		log:            logger,
+		seedClient:     seedClient,
+		shootClient:    shootClient,
+		namespace:      namespace,
+		version:        version,
+		image:          image,
+		config:         config,
+		podNetwork:     podNetwork,
+		serviceNetwork: serviceNetwork,
+	}
+}
+
+type kubeControllerManager struct {
+	log            *logrus.Entry
```

Comment:
```suggestion
	log            logrus.FieldLogger
```

Can you use logrus. FieldLogger as it is the interface that generalise `logrus.Entry` and `logrus.Logger` types. In this way we would allow `logrus.Entry` or `logrus.Logger` to be passed without forcing the caller to convert from the one to another. See https://github.com/gardener/gardener/pull/2760